### ISSUE TITLE
windows_service_status "ok" not updating 

### ISSUE THREAD

**Issue by nano4lyfe** [prometheus-community/windows_exporter#784]
Hello, 

We have installed the window_exporter on a few versions of windows now and we are seeing an issue with the windows_service_status metric - all services always point to 'ok' and we can do nothing to trigger it to change in our testing. 

If it helps, this is what we are using for our install path `msiexec /i C:\windows_exporter-0.15.0.msi ENABLED_COLLECTORS="os,service,cpu,cs,logical_disk,net,system,textfile" TEXTFILE_DIR="C:\custom" LISTEN_PORT=9100`

We were hoping to use that metric for monitoring service health but it doesn't seem to be accurate. Is there anything wrong with our install process? Or is this a known issue? I can provide more details if needed. 

Thanks

**Comment by carlpett** [comment]
Hi @nano4lyfe,
Can you specify what you mean with all services "pointing to ok"?

**Comment by nano4lyfe** [comment]
Of course, 
![image](https://user-images.githubusercontent.com/84393776/118705977-dba02a00-b7e6-11eb-854b-1703a15c77f5.png)

For example, the audio service is not started or running in any way but the status reports "ok"

![image](https://user-images.githubusercontent.com/84393776/118706112-fecad980-b7e6-11eb-97e4-b34bf91738a5.png)

We have tested by crashing/killing processes for services on the server and the exporter did not flip at all, it stayed in a constant "ok" state, it never migrated to error or stopping status even when the service was completely down. 

There could be confusion on my side but based on the metric description, I interpret it as it should be reporting if the service is in a degraded state or not or if the service is even running.

From the prometheus side: 
![image](https://user-images.githubusercontent.com/84393776/118708276-86b1e300-b7e9-11eb-8435-1853824ca32f.png)

![image](https://user-images.githubusercontent.com/84393776/118708047-3f2b5700-b7e9-11eb-8687-257008eb7a14.png)

Not exactly sure why there are two more services reported in Prometheus but I hope the above illustrates what I am trying to convey - there are 200+ services on the server and all of them are reporting ok even though maybe 50 are actually running. 

Thank you for taking the time to review this.

**Comment by carlpett** [comment]
Hm, what does `Get-WmiObject Win32_Service | ? Name -eq "audiosrv"` output for `Status`? (I'm writing this without access to a Powershell terminal, but hopefully got the line right. Apologies for any syntax errors)

**Comment by carlpett** [comment]
Oh, wait a minute - what does `windows_service_state` look like? It's somewhat confusing that there is both state and status, but the one you typically want is `_state`

**Comment by nano4lyfe** [comment]
You got the syntax 100% right. 

![image](https://user-images.githubusercontent.com/84393776/118712510-e1017280-b7ee-11eb-8208-115e0d2d2053.png)

This is the resulting output. As of right now, we have a solution in which we compare the start mode of the service and the windows service state metric to monitor this (for anyone in the future) 

`windows_service_start_mode{start_mode="auto"}  ==1 and on(instance,name)  windows_service_state{state="running",name!~ "$SEVICENAME"} != 1`

However, based on this site: https://awesome-prometheus-alerts.grep.to/rules.html

We expected the service_status metric to be sufficient to monitor a service's state. So we are confused on why this "ok" flag never changes. We aren't sure if it's a bug or expected behavior based on what the WMI object reports and it's an issue with WMI more than the exporter. 

Thanks

**Comment by carlpett** [comment]
We simply mirror what is in the WMI result, so I think this is down to understanding what Windows actually means. From the `Win32_Service` docs:
>Current status of the object. Various operational and nonoperational statuses can be defined. Operational statuses include: "OK", "Degraded", and "Pred Fail" (an element, such as a SMART-enabled hard disk drive, may be functioning properly but predicting a failure in the near future). Nonoperational statuses include: "Error", "Starting", "Stopping", and "Service". The latter, "Service", could apply during mirror-resilvering of a disk, reload of a user permissions list, or other administrative work. Not all such work is online, yet the managed element is neither "OK" nor in one of the other states.

So my interpretation would be that a service can be "ok" when stopped, if that is the desired state. This is somewhat speculative, though.

**Comment by nano4lyfe** [comment]
That is what we interpreted also when I looked up the WMI documentation. However, it seems odd since the test of killing the process should trigger the service to move to a degraded state since its desired state is running. 

So at this point, we are more pointing towards WMI having the issue and not accurately reporting the service health state than? Instead of the exporter having the issue?

 I'm curious if this is affecting other users or this is limited to our environment.
